### ISSUE TITLE
Fix flaky stratifiedImputation test

### ISSUE THREAD

**Issue by egillax** [OHDSI/PatientLevelPrediction#500]
This test fails occasionally: 
```text
── Failure ('test-featureEngineering.R:279:3'): createStratifiedImputationSettings correct class ──
nrow(stratifiedMeans) == 8 is not TRUE

`actual`:   FALSE
`expected`: TRUE 
```

**Comment by egillax** [comment]
This one is bad as well:

── Error ('test-demographicSummary.R:63:3'): getDemographicSummary ─────────────
Error in ``$<-.data.frame`(`*tmp*`, "PersonCountWithOutcome", value = numeric(0))`: replacement has 0 rows, data has 22
Backtrace:
    ▆
 1. └─PatientLevelPrediction:::getDemographicSummary_survival(...) at test-demographicSummary.R:63:3
 2.   ├─base::`$<-`(`*tmp*`, "PersonCountWithOutcome", value = `<dbl>`)
 3.   └─base::`$<-.data.frame`(`*tmp*`, "PersonCountWithOutcome", value = `<dbl>`)

**Comment by egillax** [comment]
Fixed in #510
